feat: add setting to toggle backspace key for hiding markers

dongwook-chan opened the pull request:
- Added a new setting that allows users to toggle whether the backspace key hides markers when no input is in the buffer.
- Modified the settings UI to include the new "Use Backspace to Hide Markers" toggle.
- Implemented backspace functionality that checks if the setting is enabled and hides markers accordingly.
- Updated `settings.ts` to handle this new configuration.

Closes #3.

karstenpedersen commented:
Hi, I like this. This is also the behavior of https://github.com/philc/vimium, which I inspired the plugin on.

I think it could be good if this was just the standard behavior, so both "Escape" and "Backspace" works, since it wouldn't mess with peoples current workflow.

This would remove the changes you have made in `settings.ts`. We can also change the if statement from:

`} else if (this.settings.keysForHidingMarkers.includes(event.key)) {`

to

`} else if (event.key === "Backspace" || event.key === "Escape") {`

karstenpedersen commented on src/main.ts:
```diff
@@ -32,16 +32,14 @@ export default class Vimium extends Plugin {
+				this.input = this.input.slice(0, -1);
+				this.updateMarkers();
+			}
+			else if (this.settings.keysForHidingMarkers.includes(event.key)) {
```
Hard code check

`event.key === "Escape" || event.key === "Backspace"`

karstenpedersen commented on src/settings.ts:
Discard changes in this file.

dongwook-chan committed:
feat: add setting to toggle backspace key for hiding markers
- Added a new setting that allows users to toggle whether the backspace key hides markers when no input is in the buffer.
- Modified the settings UI to include the new "Use Backspace to Hide Markers" toggle.
- Implemented backspace functionality that checks if the setting is enabled and hides markers accordingly.
- Updated `settings.ts` to handle this new configuration.

dongwook-chan commented:
@karstenpedersen 
I rollbacked the code and made changes according to your review and then forced-pushed. PTAL

karstenpedersen committed:
Update main.ts

karstenpedersen commented:
Thanks for the pull request :)
